javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -776,8 +776,9 @@ private Path normalizePath(Path path) {
     return path.toAbsolutePath().normalize();
   }
 
-  private void extractTypeTable(Path fileHandle, TypeTable table) {
-    TrapWriter trapWriter = outputConfig.getTrapWriterFactory().mkTrapWriter(fileHandle.toFile());
+  private void extractTypeTable(TypeTable table) {
+    TrapWriter trapWriter =
+        outputConfig.getTrapWriterFactory().mkTrapWriter(new File(".codeql-typescript-typetable"));
```

Comment:
Does this code assume that there is only ever going to be one type table? Is that a safe assumption?

Reply:
There is one type table per TypeScript extractor invocation.

So we might have a problem if someone is using odasa to build a project consisting of multiple extractor invocations all in the same base directory. But if extractor invocations can overwrite each others' trap files, I think we already had that problem. Short of using random strings, I'm not sure how to name a trap file so it can't get clobbered?

Reply:
Could we derive the path of the trap file from the path of the `tsconfig.json` file, perhaps by appending the magic `.codeql-typescript-typetable` string to it?

Reply:
Yes, although different extractor invocations can in theory include the same `tsconfig.json` file (but have different overall inclusion/exclusion patterns so they don't extract the same set of types).

Reply:
Fair enough, but that still seems slightly safer than always using a constant string?